python/hail/methods/statgen.py

```diff
@@ -204,6 +204,729 @@ def linreg(dataset, ys, x, covariates=[], root='linreg', block_size=16):
 
     return cleanup(MatrixTable(jm))
 
+@handle_py4j
+@typecheck(dataset=MatrixTable,
+           test=strlike,
+           y=Expression,
```

Comment:
do these have to be numeric?

Reply:
I think there's a typechecker for that, numeric_expr

Reply:
@tpoterba Do you have any thoughts on the appropriate typecheck here and `linreg` and `lmmreg`? Some complications are that `linreg` can take a single numeric expression or a list of them for `ys`, and `lmmreg` can take a boolean or a numeric. I tried using `NumericExpression` and `AtomicExpression`, since typechecking against `Expression` matches subclasses, but that doesn't appear to work.

Reply:
I think here you should do `expr_numeric`

Reply:
But `y` might be a boolean expression. And what about `linreg/ys`?

Reply:
ah, in that case `oneof(expr_bool, expr_numeric)`.

for linreg, probably `oneof(listof(oneof(expr_bool, expr_numeric)), expr_bool, expr_numeric)`

That's kinda ugly but probably the best thing right now.

Reply:
Thanks, I wasn't sure if the `expr_foo` checkers that do automatic captures worked inside the higher-order checkers.